app/app.js

```diff
@@ -71,6 +71,17 @@ module.exports = (options) => {
   // Handle the banner component serverside.
   require('./banner.js')(app)
 
+  // Define middleware for all routes
+  app.use('*', function (request, response, next) {
+    app.locals.legacy = request.query['legacy'] === '1'
```

Comment:
I think we want these to be [`res.locals`](https://expressjs.com/en/api.html#res.locals) as this is scoped to the specific request that's being made.